src/wasm.cc

```diff
@@ -546,9 +590,8 @@ getOrCreateThreadLocalWasm(const std::shared_ptr<WasmHandleBase> &base_handle,
     if (wasm_handle) {
       return wasm_handle;
     }
-    // Remove stale entry.
-    local_wasms.erase(vm_key);
   }
+  removeStaleLocalCacheEntries(local_wasms, local_wasms_keys);
```

Comment:
Nitpicking, but we know that `vm_key` has a stale entry at this point, and it might not be removed in `removeStaleLocalCacheEntries` due to the iteration limit, so it would be good to erase it inside the `if` block anyway.

Reply:
Sure, done 👌